ansible/roles/letsencrypt/tasks/main.yml

```diff
@@ -33,46 +18,32 @@
     - "/etc/nginx/sites-enabled"
     - "/var/www/letsencrypt"
 
-- name: check for /.well-known/acme-challenge
-  shell: "grep '/.well-known/acme-challenge' /etc/nginx/sites-enabled/*"
-  changed_when: false
-  ignore_errors: yes
-  register: acme_challenge_present
-
-- name: Add letsencrypt vhost
+- name: Add letsencrypt dedicated sites-enabled file
   template:
     src: templates/letsencrypt-http
     dest: /etc/nginx/sites-enabled/letsencrypt-http
+    mode: 0644
+    owner: root
   notify: reload nginx
-  when: "acme_challenge_present.rc != 0"
 
 - meta: flush_handlers
 
-# NB: --standalone is not supported
-- name: Generate Letsencrypt certificate with certbot package
+# Run certbot to create a new cert if needed
+# It's one certificate for all the domains in letsencrypt_domains
+# named after the first domain
+# If new domains have been added certbot updates the cert (--expand)
+# otherwise it keeps the existing cert (--keep)
+- name: Create Letsencrypt certificates for the first time
+  args:
   command: >
     certbot certonly
     --webroot --webroot-path /var/www/letsencrypt
     --preferred-challenges http
     --noninteractive --text --agree-tos
     --email {{ letsencrypt_admin_email }}
     --domains {{ letsencrypt_domains | join(',') }}
-    {% if expand is defined %}--expand{% endif %}
+    --expand
+    --keep
```

Comment:
Are you sure we should not be using the `--nginx` option for nginx?

Reply:
This questions/comment still is relevant I think.

Reply:
Certbot is writing the challenge in a directory and calling the post hook  " --post-hook '/usr/sbin/service nginx reload'" during cert updates.
It's not using the Nginx plugin anymore and it's unaware of Nginx in itself.